fix: also rename UserId to userId in method body

kesselb opened the pull request:
<!--- Provide a general summary of your changes in the title above. -->

## Description
<!--- Describe your changes in detail. -->

Follow-up for https://github.com/nextcloud-libraries/rector/pull/7

## Motivation and context
<!--- Why is this change required? What problem does it solve? -->
<!--- If it fixes an open issue, please link to the issue here. -->

It seems reasonable to rename `UserId`  to `userId`  in the method body as well :rofl: 

Note: That only works for assignments. Need some more code to also replace it in function calls. 

## How has this been tested?
<!--- Please describe in detail how you tested your changes. -->
<!--- Include details of your testing environment, and the tests you ran to -->
<!--- see how your change affects other areas of the code, etc. -->

## Types of changes
<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## PR checklist
<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING.md** document.
- [x] I have added tests to cover my changes.

ChristophWurst commented on src/Rector/RenameUserIdRector.php:
```diff
@@ -74,6 +77,20 @@ public function refactor(Node $node): ?Node
 
+        $stmts = $constructClassMethod->getStmts();
+
+        if (is_array($stmts)) {
+            foreach ($stmts as $stmt) {
+                if ($stmt instanceof Expression) {
+                    if ($stmt->expr instanceof Assign) {
+                        if ($stmt->expr->expr instanceof Variable && $stmt->expr->expr->name === 'UserId') {
+                            $stmt->expr->expr->name = 'userId';
+                        }
+                    }
+                }
+            }
+        }
```
How about early returns and early continues?

come-nc commented:
Can you make it generic so that we use it for AppName and WebRoot as well?
ref: https://docs.nextcloud.com/server/latest/developer_manual/app_publishing_maintenance/app_upgrade_guide/upgrade_to_26.html#dependency-injection-parameters

kesselb commented:
> Can you make it generic so that we use it for AppName and WebRoot as well?

Good idea!

- Renamed to RenameParameterRector
- Added configuration objects
- Also replace the variables in method and static calls

kesselb commented:
I have another follow-up in the queue, better detection of extended classes with reflection resolver.
However, psalm doesn't allow me to push the change because the constructor does not initialize all properties :see_no_evil: 

![image](https://github.com/user-attachments/assets/dbaa7388-94fa-4f4a-bdd4-4fa5f9fd9b3b)

come-nc commented:
> I have another follow-up in the queue, better detection of extended classes with reflection resolver. However, psalm doesn't allow me to push the change because the constructor does not initialize all properties 🙈
> 
> ![image](https://private-user-images.githubusercontent.com/3902676/368763593-dbaa7388-94fa-4f4a-bdd4-4fa5f9fd9b3b.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjY3MDM1OTIsIm5iZiI6MTcyNjcwMzI5MiwicGF0aCI6Ii8zOTAyNjc2LzM2ODc2MzU5My1kYmFhNzM4OC05NGZhLTRmNGEtYmRkNC00ZmE1ZjlmZDliM2IucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDkxOCUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA5MThUMjM0ODEyWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9ODM5ZjM2YmVhNDEwMDBiMGJmOGEzN2Q3ODllNjdmOTdiNzY0OThkMjg2YmY3MTM1ZGUyNTRhNjQxMWY5ZWQ5ZSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.EalWZfa-hdGvahe_7GDYV7OzFEFD04MkQlf2ibOSx5M)

Initialize them, or make them nullable and init to null?

come-nc commented:
> The base branch requires all commits to be signed. Learn more about signing commits.

@kesselb

kesselb commented:
Done ;)

Note: The current version can not replace in if/foreach/etc.

come-nc commented:
> Done ;)
> 
> Note: The current version can not replace in if/foreach/etc. If you think the rector is too risky, then we should not enable it by default.

I’d say it’s fine, constructor are usually simple, and if it misses one CI should spot the problem quite easily.
